app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java

```diff
@@ -235,15 +246,22 @@ private void monitorSubscription(final ChannelInfo info) {
                 .subscribe(getSubscribeUpdateMonitor(info), onError));
 
         disposables.add(observable
-                // Some updates are very rapid
-                // (for example when calling the updateSubscription(info))
-                // so only update the UI for the latest emission
-                // ("sync" the subscribe button's state)
-                .debounce(100, TimeUnit.MILLISECONDS)
+                .map(List::isEmpty)
+                .distinctUntilChanged()
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribe((List<SubscriptionEntity> subscriptionEntities) ->
-                        updateSubscribeButton(!subscriptionEntities.isEmpty()), onError));
+                .subscribe((Boolean isEmpty) -> updateSubscribeButton(!isEmpty), onError));
 
+        disposables.add(observable
+                .map(List::isEmpty)
+                .filter(x -> NotificationHelper.isNewStreamsNotificationsEnabled(requireContext()))
+                .distinctUntilChanged()
```

Comment:
What about inverting the position of these two lines, so that `isNewStreamsNotificationsEnabled()` is not continuously queried?

Reply:
done